gengokit/httptransport/httptransport.go

```diff
@@ -260,9 +260,19 @@ err = json.Unmarshal([]byte({{.LocalName}}Str), {{.LocalName}})`
 		// All repeated args of any type are represented as slices, and bare
 		// assignments to a slice accept a slice as the rvalue. As a result,
 		// LocalName will be declared as a slice, and json.Unmarshal handles
-		// everything else for us.
+		// everything else for us. Addititionally, if a type is a Base type and
+		// is repeated, we first attempt to unmarshal the string we're
+		// provided, and if that fails, we try to unmarshal the string
+		// surrounded by square brackets. If THAT fails, then the string does
+		// not represent a valid JSON string and an error is returned.
 		repeatedUnmarshalTmpl := `
 var {{.LocalName}} {{.GoType}}
+{{- if and (and .IsBaseType .Repeated) (not (Contains .GoType "[]byte"))}}
+err = json.Unmarshal([]byte({{.LocalName}}Str), &{{.LocalName}})
```

Comment:
Okay guys, I got way too involved here and decided to try some science. Well, pseudoscience. I made up some benchmarks, with bogus input data, to try to suss out the best of a few approaches.

Check out the results here: https://github.com/ianfoo/go-json-unmarshal-experiment

Long story short, it looks like using a heuristic to determine whether the input needs to be wrapped with square brackets is the fastest and requires the least allocations. There may possibly be a better approach, but out of the ones we've talked about, checking and adjusting, then unmarshaling, wins.

This said, the boolean check in this thread may not be what we want. Check the results here: https://play.golang.org/p/tnB3U_c-ua. The boolean for checking square brackets here will indicate that an input does not need to be wrapped if *either* begins or ends with a square bracket (or both). The one that I have used will only indicate that an input needs to be wrapped if it begins *and* ends with square brackets.